geode-core/src/main/java/org/apache/geode/internal/cache/DistributedRegion.java

```diff
@@ -181,6 +182,7 @@
    * Lock to prevent multiple threads on this member from performing a clear at the same time.
    */
   private final Object clearLock = new Object();
+  private final ReentrantReadWriteLock failedGIIClearLock = new ReentrantReadWriteLock(true);
```

Comment:
In this case, it doesn't look like we need to have fair lock; as there are only few write lock to be acquired, during failed clear. How about using StoppableReentrantReadWriteLock which takes cancel criteria.